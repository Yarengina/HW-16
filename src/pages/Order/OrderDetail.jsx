// eslint-disable-next-line no-unused-vars
import * as React from 'react'
import { Link } from 'react-router-dom'
import Counter from './Counter'

function OrderDetail({ books, setQuantity }) {
    return (
        <div className="BookField">
            {books.map(({ id, title, price, rest, quantity }) => (
                <div data-testid={`todo-${id}`} className="Book" key={id}>
                    <p className="Title">{title}</p>
                    <p className="Price">{price}</p>
                    <Link className="ReadMoreLink" to={`${id}`}>
                        Read more
                    </Link>
                    <Counter
                        min={1}
                        max={rest + 1}
                        current={quantity}
                        onChange={(bookQuantity) =>
                            setQuantity(id, bookQuantity)
                        }
                    />
                </div>
            ))}
        </div>
    )
}

export default OrderDetail
