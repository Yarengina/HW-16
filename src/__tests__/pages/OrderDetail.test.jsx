// eslint-disable-next-line no-unused-vars
import * as React from 'react'
import '@testing-library/jest-dom/extend-expect'
import { screen } from '@testing-library/react'
import { render } from '../utils/renderWithRouter'
import OrderDetail from '../../pages/Order/OrderDetail'

describe('<OrderDetail />', () => {
    const books = [
        {
            id: 101,
            title: 'Война и мир - Л.Н.Толстой',
            description:
                'Роман-эпопея Льва Николаевича Толстого, описывающий русское общество в эпоху войн против Наполеона в 1805—1812 годах. Эпилог романа доводит повествование до 1820 года.',
            price: 800,
            rest: 10,
            quantity: 1,
        },
    ]

    it('should render book detail with id 101', () => {
        render(<OrderDetail books={books} />)
        const bookElement = screen.getByTestId('todo-101')
        expect(bookElement).toBeInTheDocument()
        expect(bookElement).toHaveTextContent('Война и мир - Л.Н.Толстой')
        expect(bookElement).toHaveTextContent(800)
        expect(bookElement).toContainHTML('href="/101"')
    })

    it('should render successfully and match snapshot', () => {
        const { container } = render(<OrderDetail books={books} />)
        expect(container).toMatchSnapshot()
    })
})
