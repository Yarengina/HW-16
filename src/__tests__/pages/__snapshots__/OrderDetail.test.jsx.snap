// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<OrderDetail /> should render successfully and match snapshot 1`] = `
<div>
  <div
    class="BookField"
  >
    <div
      class="Book"
      data-testid="todo-101"
    >
      <p
        class="Title"
      >
        Война и мир - Л.Н.Толстой
      </p>
      <p
        class="Price"
      >
        800
      </p>
      <a
        class="ReadMoreLink"
        href="/101"
      >
        Read more
      </a>
      <div
        class="Counter"
      >
        <button
          class="Buttons Plus"
          type="button"
        >
          Remove item
        </button>
        <p
          class="Quantity"
        >
          0
        </p>
        <button
          class="Buttons Minus"
          type="button"
        >
          Add to cart
        </button>
      </div>
    </div>
  </div>
</div>
`;
